### ISSUE TITLE
Fixed product tax support

### ISSUE THREAD

**Issue by Amadeco** [OCA/connector-magento#170]
Good evening,

It is very difficult to find a good tutorial, and answers to install the python module "connector-magento" but also the magento module "Openlabs_OpenERPConnector" which miss a call action in the api (ol.catalog.info where you find it in a old bz branch launchpad, whoa)

app/code/community/Openlabs/OpenERPConnector
<ol_catalog_product>, Line 409

```
                    <info translate="title" module="core">
                        <title>Retrieve product</title>
                        <method>info</method>
                    </info>
```

app/code/community/Openlabs/OpenERPConnector/Model/Olcatalog/Products.php
Openlabs_OpenERPConnector_Model_Olcatalog_Products

```
public function info($productId, $store = null, $attributes = null, $identifierType = null)
    {
        $result = Mage::getSingleton('catalog/product_api')->info($productId, $store, $attributes, $identifierType);
        $resultObject = new Varien_Object($result);
        $type = $result['type'];

        $resultObject->setData('description', strip_tags($resultObject->getData('description')));

        Mage::dispatchEvent('ol_openerp_catalog_product_info_type_'.$type, array(
                'result_object' => $resultObject, 'product_id' => $productId, 'store' => $store));

        return $resultObject->getData();
    }
```

We downgrade to odoo version 8, we didn't success to install it in the last version (9.0.0)
We finally success to download catalog product (not perfect informations), catalog categories (do not import categories and products separatly, it cause to download the same category tree twice) and customers (I do not know why but a few customers are assignated to company ?)

Orders without success. It seems a problem with the API (invalid ressource path).

Also, You can not update products from Magento once downloaded by Odoo.

_But the point I want to ask, is how do you manage your weee tax between magento and odoo ?_
You need to download Fixed Product Tax by products to be perfect.

Great product otherwise, open-source, especially free for a bundle of apps perfect to CRM to Fulfillment but lacks of documentation.

Cheers

**Comment by guewen** [comment]
What was missing on the documentation that could have helped you? http://odoo-magento-connector.com/guides/installation_guide.html

The addons have not been migrated to Odoo 9.0 as most of the community addons. So indeed you can't install it on Odoo 9.0. There is a great amount of work needed for that. 

I'm very very surprised about the missing method, which launchpad branch did you use? The one from the documentation?

Products changed on Magento _are_ updated on Odoo. You have to run manually an import of the products on the back end or activate the "Magento - Import products" cron job. The Import is a differential from the last import.

For the category tree and partners, please open separates bug reports with more details.

Lastly, regarding the taxes, as the taxes in Magento and Odoo are very different, you have to configure them in both places (usually with default values in Odoo). If you misconfigured a tax on a product anyway a sales order using it will be blocked until the issue is resolved.

**Comment by azeemism** [comment]
`"The addons have not been migrated to Odoo 9.0 as most of the community addons. So indeed you can't install it on Odoo 9.0. There is a great amount of work needed for that."`

@guewen does this mean that odoo 9.0 cannot be linked to magento 1.9.2.2  currently using http://odoo-magento-connector.com/guides/installation_guide.html if `-b 9.0` was cloned for

`connector`
`connector-ecommerce`
`connector-magento`
`e-commerce`
`product-attribute`
`sale-workflow`
`server-tools`

All these dependencies look to default to branch 9.0, so I am a little confused.

Do you mean that 
`$ bzr branch lp:magentoerpconnect/magento-module-oerp6.x-stable magento-module`
needs to be updated before it will work with odoo 9.0

Is the readme for connector-magento not accurate...
`"Odoo Magento Connector (also known as MagentoERPconnect) is a bi-directional connector, 100% compatible with the last OpenERP 9.0 and latest Magento versions (1.x)."`

For a current production installation, would you suggest an alternate connector be used to connect odoo 9.0 and magento 1.9.2.2 if one is available?

Thanks

**Comment by guewen** [comment]
> All these dependencies look to default to branch 9.0, so I am a little confused.

Yes but all the addons in there have their `installable` flag as `False`.

> Do you mean that
> $ bzr branch lp:magentoerpconnect/magento-module-oerp6.x-stable magento-module
> needs to be updated before it will work with odoo 9.0

No because this module is on the Magento side. Still it is a wish of us to have a new clean version of this extension.

> For a current production installation, would you suggest an alternate connector be used to connect odoo 9.0 and magento 1.9.2.2 if one is available?

I don't know.

**Comment by azeemism** [comment]
@guewen thanks for clearing this up. Looks like it will be best to use connector-magento
 with Odoo 8.0 from
`git clone https://www.github.com/OCA/OCB.git --depth 1 --branch 8.0 --single-branch .`
and Magento 1.9.2.2. Using branch 8.0 should also make the other OCA addons installable.

Appreciate all your hard work.
